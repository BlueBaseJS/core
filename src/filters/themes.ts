import { BlueBase, BootOptions } from '../BlueBase';

import { BlueBaseTheme } from '../themes';
import { FilterNestedCollection } from '../registries';

export const themes: FilterNestedCollection = {
	'bluebase.themes.register': [
		/**
		 * Registers all themes that ship with BlueBase
		 */
		{
			key: 'bluebase-themes-register-internal-default',
			priority: 2,

			value: async (bootOptions: BootOptions, _ctx: {}, BB: BlueBase) => {
				const keys = await BB.Themes.registerCollection([BlueBaseTheme]);
				keys.forEach((key: string) => BB.Themes.setMeta(key, 'source', { type: 'system' }));

				return bootOptions;
			},
		},

		/**
		 * Registers all themes in the bluebase.ts
		 */
		{
			key: 'bluebase-themes-register-default',
			priority: 3,

			value: async (bootOptions: BootOptions, _ctx: {}, BB: BlueBase) => {
				const keys = await BB.Themes.registerCollection(bootOptions.themes);
				keys.forEach((key: string) => BB.Themes.setMeta(key, 'source', { type: 'boot' }));

				return bootOptions;
			},
		},

		/**
		 * Registers all themes in the plugins
		 */
		{
			key: 'bluebase-themes-register-from-plugins',
			priority: 4,

			value: async (bootOptions: BootOptions, _ctx: {}, BB: BlueBase) => {
				const plugins = await BB.Plugins.getAllEnabled();

				for (const plugin of plugins) {
					const keys = await BB.Themes.registerCollection(plugin.themes);
					keys.forEach((key: string) =>
						BB.Themes.setMeta(key, 'source', { type: 'plugin', key: plugin.key })
					);
				}

				return bootOptions;
			},
		},
	],
};
