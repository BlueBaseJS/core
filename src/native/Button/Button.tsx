import React from 'react';

const {
	Platform,
	StyleSheet,
	Text,
	TouchableNativeFeedback,
	TouchableOpacity,
	View
// tslint:disable-next-line:no-var-requires
} = require('react-native');

export type ButtonProps = {
  /**
   * Text to display inside the button
   */
	children: React.ReactNode,

		/**
		 * Handler to be called when the user taps the button
		 */
		onPress: (event?: any) => any,

			/**
			 * Color of the text (iOS), or background color of the button (Android)
			 */
			color?: string,

			/**
			 * TV preferred focus (see documentation for the View component).
			 */
			hasTVPreferredFocus?: boolean,

			/**
			 * Text to display for blindness accessibility features
			 */
			accessibilityLabel?: string,

			/**
			 * If true, disable all interactions for this component.
			 */
			disabled?: boolean,

			/**
			 * Used to locate this view in end-to-end tests.
			 */
			testID?: string,
};

/**
 * A basic button component that should render nicely on any platform. Supports
 * a minimal level of customization.
 *
 * If this button doesn't look right for your app, you can build your own
 * button using [TouchableOpacity](docs/touchableopacity.html)
 * or [TouchableNativeFeedback](docs/touchablenativefeedback.html).
 *
 * Example usage:
 *
 * ```
 * import { Button } from 'react-native';
 * ...
 *
 * <Button
 *   onPress={onPressLearnMore}
 *   color="#841584"
 *   accessibilityLabel="Learn more about this purple button"
 * />
 * Learn More
 * </Button>
 * ```
 *
 */

export class Button extends React.Component<ButtonProps> {
	render() {
		const {
			accessibilityLabel,
			color,
			onPress,
			children,
			hasTVPreferredFocus,
			disabled,
			testID,
		} = this.props;
		const buttonStyles = [styles.button];
		const textStyles = [styles.text];
		if (color) {
			if (Platform.OS === 'ios') {
				textStyles.push({ color: color });
			} else {
				buttonStyles.push({ backgroundColor: color });
			}
		}
		const accessibilityStates = [];
		if (disabled) {
			buttonStyles.push(styles.buttonDisabled);
			textStyles.push(styles.textDisabled);
			accessibilityStates.push('disabled');
		}
		const Touchable =
			Platform.OS === 'android' ? TouchableNativeFeedback : TouchableOpacity;
		return (
			<Touchable
				accessibilityLabel={accessibilityLabel}
				accessibilityRole="button"
				accessibilityStates={accessibilityStates}
				hasTVPreferredFocus={hasTVPreferredFocus}
				testID={testID}
				disabled={disabled}
				onPress={onPress}>
				<View style={buttonStyles}>
					{typeof children === 'string'
						? <Text style={textStyles} disabled={disabled}>{children}</Text>
						: children
					}
				</View>
			</Touchable>
		);
	}
}

const styles = StyleSheet.create({
	button: Platform.select({
		ios: {},
		android: {
			elevation: 4,
			// Material design blue from https://material.google.com/style/color.html#color-color-palette
			backgroundColor: '#2196F3',
			borderRadius: 2,
		},
		web: {
			// Material design blue from https://material.google.com/style/color.html#color-color-palette
			backgroundColor: '#2196F3',
			borderRadius: 2,
		},
	}),
	text: Platform.select({
		ios: {
			// iOS blue from https://developer.apple.com/ios/human-interface-guidelines/visual-design/color/
			color: '#007AFF',
			textAlign: 'center',
			padding: 8,
			fontSize: 18,
		},
		android: {
			color: 'white',
			textAlign: 'center',
			padding: 8,
			fontWeight: '500',
		},
		web: {
			color: 'white',
			textAlign: 'center',
			padding: 8,
			fontWeight: '500',
		},
	}),
	buttonDisabled: Platform.select({
		ios: {},
		android: {
			elevation: 0,
			backgroundColor: '#dfdfdf',
		},
	}),
	textDisabled: Platform.select({
		ios: {
			color: '#cdcdcd',
		},
		android: {
			color: '#a1a1a1',
		},
	}),
});
