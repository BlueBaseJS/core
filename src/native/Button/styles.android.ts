import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	button: {
		// Material design blue from https://material.google.com/style/color.html#color-color-palette
		backgroundColor: '#2196F3',
		borderRadius: 2,
		elevation: 4,
	},
	buttonDisabled: {
		backgroundColor: '#dfdfdf',
		elevation: 0,
	},
	text: {
		color: 'white',
		fontWeight: '500',
		padding: 8,
		textAlign: 'center',
	},
	textDisabled: {
		color: '#a1a1a1',
	},
});