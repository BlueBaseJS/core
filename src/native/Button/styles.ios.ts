import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	button: {},
	buttonDisabled: {},
	text: {
		// iOS blue from https://developer.apple.com/ios/human-interface-guidelines/visual-design/color/
		color: '#007AFF',
		fontSize: 18,
		padding: 8,
		textAlign: 'center',
	},
	textDisabled: {
		color: '#cdcdcd',
	},
});