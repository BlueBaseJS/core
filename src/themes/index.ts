export * from './helpers';
export * from './structure';
export * from './themes';
export * from './ThemeContext';
