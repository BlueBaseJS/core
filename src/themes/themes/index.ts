export * from './BlueBaseDarkTheme';
export * from './BlueBaseLightTheme';
