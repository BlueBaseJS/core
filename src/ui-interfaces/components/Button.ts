import React from 'react';

export interface ButtonProps {

 /* Label to be passed as child. */
	children?: React.ReactNode;

	/** Style object for button's style. */
	style?: object; // TODO: do proper typings

	/* Color prop of type enum. */
	color?:
		| 'primary'
		| 'secondary'
		| 'warning'
		| 'info'
		| 'danger'
		| 'success'
		| 'light'
		| 'dark'
		| 'link'
		| 'accent'
		| 'default'
		| 'contrast'
		| 'inherit'
		| 'white';

	/**
	 * Callback function fired when button is pressed.
	 */
	onPress?: () => void;

	/**
	 * If true, renders a disabled button.
	 */
	disabled?: boolean;

	/**
	 * Supportive for Web only, href prop to move to a link.
	 */
	href?: string;

	/**
	 * If true, button is generated with 100% width of the cnotainer.
	 */
	fullWidth?: boolean;

	/**
	 * If true, shows active state of the button.
	 */
	active?: boolean;

	/**
	 * If true, button is generated with border, unsupportive for MaterialUI button.
	 */
	bordered?: boolean;

	/**
	 * If true, button is generated rounded (Native only).
	 */
	rounded?: boolean;

	/**
	 * Icon prop to show icon on the left side of label.
	 */
	iconLeft?: React.ReactNode;

	/**
	 * Icon prop to show icon on the right side of label.
	 */
	iconRight?: React.ReactNode;

	/**
	 * Fab prop to show floating action button, unsupportive for Bootstrap Button.
	 */
	fab?: boolean;

	/**
	 * If true, small button is generated (For MaterialUI button, small also works when fab is true).
	 */
	small?: boolean;

	/**
	 * If true, large button is generated. If both small and large are true then Large button is generated.
	 */
	large?: boolean;

	[key: string]: any;
}