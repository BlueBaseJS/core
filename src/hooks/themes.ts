import { BlueBase, BootOptions } from '../BlueBase';
import { BlueBaseDarkTheme, BlueBaseLightTheme } from '../themes';
import { HookInput } from '../registries';

export const themes: { [key: string]: HookInput[] } = {

	'bluebase.themes.register': [{
		name: 'bluebase-components-register-internal-default',
		priority: 5,

		// tslint:disable-line:object-literal-sort-keys
		handler: async (bootOptions: BootOptions, _ctx: {}, BB: BlueBase) => {

			// TODO: Use .register method instead, as themes may be split
			BB.Themes.set('BlueBase.Light', BlueBaseLightTheme);
			BB.Themes.set('BlueBase.Dark', BlueBaseDarkTheme);
			return bootOptions;
		},
	}]
};
