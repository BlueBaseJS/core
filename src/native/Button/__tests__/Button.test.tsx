import { Button, ButtonStyles } from '../Button';

import { BlueBaseTheme } from '../../../themes';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Text } from 'react-native';
import { mount } from 'enzyme';
import { waitForElement } from 'enzyme-async-helpers';

describe('Button', () => {
	test(`should render Button with default color styles`, () => {
		const styles = Button.defaultStyles(BlueBaseTheme) as ButtonStyles;
		const rendered = TestRenderer.create(
			<Button styles={styles} testID="button-root">
				Hello
			</Button>
		);

		const el = rendered.root.findAllByProps({ testID: 'button-root' })[1];
		const text = rendered.root.findByType(Text);

		expect(el.props.style[0]).toMatchObject(styles.root);
		expect(el.props.style[1]).toMatchObject(styles.default);

		expect(text.props.style[0]).toMatchObject(styles.text);
		expect(text.props.style[0]).toMatchObject(styles.defaultText);
	});

	test(`should render Button with primary color styles`, () => {
		const styles = Button.defaultStyles(BlueBaseTheme) as ButtonStyles;
		const rendered = TestRenderer.create(
			<Button styles={styles} color="primary" testID="button-root">
				Hello
			</Button>
		);

		const el = rendered.root.findAllByProps({ testID: 'button-root' })[1];
		const text = rendered.root.findByType(Text);

		expect(el.props.style[0]).toMatchObject(styles.root);
		expect(el.props.style[1]).toMatchObject(styles.primary);

		expect(text.props.style[0]).toMatchObject(styles.text);
		expect(text.props.style[1]).toMatchObject(styles.primaryText);
	});

	test(`should render Button with secondary color styles`, () => {
		const styles = Button.defaultStyles(BlueBaseTheme) as ButtonStyles;
		const rendered = TestRenderer.create(
			<Button styles={styles} color="secondary" testID="button-root">
				Hello
			</Button>
		);

		const el = rendered.root.findAllByProps({ testID: 'button-root' })[1];
		const text = rendered.root.findByType(Text);

		expect(el.props.style[0]).toMatchObject(styles.root);
		expect(el.props.style[1]).toMatchObject(styles.secondary);

		expect(text.props.style[0]).toMatchObject(styles.text);
		expect(text.props.style[1]).toMatchObject(styles.secondaryText);
	});

	test(`should render Button with link color styles`, () => {
		const styles = Button.defaultStyles(BlueBaseTheme) as ButtonStyles;
		const rendered = TestRenderer.create(
			<Button styles={styles} color="link" testID="button-root">
				Hello
			</Button>
		);

		const el = rendered.root.findAllByProps({ testID: 'button-root' })[1];
		const text = rendered.root.findByType(Text);

		expect(el.props.style[0]).toMatchObject(styles.root);
		expect(el.props.style[1]).toMatchObject(styles.link);

		expect(text.props.style[0]).toMatchObject(styles.text);
		expect(text.props.style[1]).toMatchObject(styles.linkText);
	});

	test(`should render Button with full width`, () => {
		const styles = Button.defaultStyles(BlueBaseTheme) as ButtonStyles;
		const rendered = TestRenderer.create(
			<Button styles={styles} testID="button-root" fullWidth={true}>
				Hello
			</Button>
		);

		const el = rendered.root.findAllByProps({ testID: 'button-root' })[1];
		const text = rendered.root.findByType(Text);

		expect(el.props.style[0]).toMatchObject(styles.root);
		expect(el.props.style[1]).toMatchObject(styles.default);
		expect(el.props.style[2]).toMatchObject(styles.fullWidth);

		expect(text.props.style[0]).toMatchObject(styles.text);
		expect(text.props.style[1]).toMatchObject(styles.defaultText);
	});

	test(`should render Button with custom children`, () => {
		const styles = Button.defaultStyles(BlueBaseTheme) as ButtonStyles;
		const rendered = TestRenderer.create(
			<Button styles={styles} testID="button-root" fullWidth={true}>
				<Text testID="custom">Custom</Text>
			</Button>
		);

		const el = rendered.root.findAllByProps({ testID: 'button-root' })[1];
		const text = rendered.root.findByType(Text);

		expect(el.props.style[0]).toMatchObject(styles.root);
		expect(el.props.style[1]).toMatchObject(styles.default);
		expect(el.props.style[2]).toMatchObject(styles.fullWidth);

		expect(text.props.style).toBeUndefined();
		expect(text.props.testID).toBe('custom');
	});

	test(`should use TouchableNativeFeedback on android platform`, async () => {
		jest.mock('Platform', () => {
			const Platform = (require as any).requireActual('Platform');
			Platform.OS = 'android';
			return Platform;
		});

		const styles = Button.defaultStyles(BlueBaseTheme) as ButtonStyles;

		// const rendered = TestRenderer.create(
		// 	<Button styles={styles}>
		// 		Hello
		// 	</Button>
		// );

		// // const el = rendered.root.findAllByProps({ testID: 'button-root' })[1];
		// const touchable = rendered.root.findByType(TouchableNativeFeedback);

		// expect(touchable.type).toBe('TouchableNativeFeedback');

		const wrapper = mount(<Button styles={styles}>Hello</Button>);

		// Wait for render
		await waitForElement(wrapper as any, Button);

		expect(wrapper.find('Button DummyTouchableNativeFeedback').length).toBe(1);
	});
});
