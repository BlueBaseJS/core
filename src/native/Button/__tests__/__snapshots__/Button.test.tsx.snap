// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button with custom children 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderRadius": 4,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
        Object {
          "backgroundColor": "#fafafa",
        },
        Object {
          "alignSelf": "stretch",
        },
      ]
    }
    testID="button-root"
  >
    <Text
      testID="custom"
    >
      Custom
    </Text>
  </View>
</View>
`;

exports[`Button should render Button with default color styles 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderRadius": 4,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
        Object {
          "backgroundColor": "#fafafa",
        },
      ]
    }
    testID="button-root"
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {},
        ]
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`Button should render Button with full width 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderRadius": 4,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
        Object {
          "backgroundColor": "#fafafa",
        },
        Object {
          "alignSelf": "stretch",
        },
      ]
    }
    testID="button-root"
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {},
        ]
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`Button should render Button with link color styles 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderRadius": 4,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
        Object {},
      ]
    }
    testID="button-root"
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {},
        ]
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`Button should render Button with primary color styles 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderRadius": 4,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
        Object {
          "backgroundColor": "#3f51b5",
        },
      ]
    }
    testID="button-root"
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`Button should render Button with secondary color styles 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderRadius": 4,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
        Object {
          "backgroundColor": "#f50057",
        },
      ]
    }
    testID="button-root"
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`Button should use TouchableNativeFeedback on android platform 1`] = `
<Button
  styles={
    Object {
      "default": Object {
        "backgroundColor": "#fafafa",
      },
      "defaultText": Object {},
      "fullWidth": Object {
        "alignSelf": "stretch",
      },
      "link": Object {},
      "linkText": Object {},
      "primary": Object {
        "backgroundColor": "#3f51b5",
      },
      "primaryText": Object {
        "color": "#fff",
      },
      "root": Object {
        "alignSelf": "flex-start",
        "borderRadius": 4,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      "secondary": Object {
        "backgroundColor": "#f50057",
      },
      "secondaryText": Object {
        "color": "#fff",
      },
      "text": Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontWeight": "500",
        "textAlign": "center",
        "textTransform": "uppercase",
      },
    }
  }
>
  <DummyTouchableNativeFeedback
    accessibilityRole="button"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "margin": 20,
              }
            }
          >
            TouchableNativeFeedback is not supported on this platform!
          </Text>
        </Text>
      </View>
    </View>
  </DummyTouchableNativeFeedback>
</Button>
`;
